runner/client/src/mill/runner/client/MillProcessLauncher.java

```diff
@@ -248,15 +250,81 @@ static int getTerminalDim(String s, boolean inheritError) throws Exception {
 
   private static AtomicReference<String> memoizedTerminalDims = new AtomicReference();
 
+  private static final boolean canUseNativeTerminal;
+
+  static {
```

Comment:
This block is pretty messy, and I'm wondering if there's some way we can make it obvious that the fast path is fast and does not classload any heavy libraries. That is the case now, but the fact that it's not obvious from the code makes it likely to regress accidentally if people touch this code.

Perhaps we could do something like what we do for the JVM versions in https://github.com/com-lihaoyi/mill/blob/392f32d9f3d7e7cb4a4873a1f6c5c337dc63432b/runner/client/src/mill/runner/client/MillProcessLauncher.java#L134-L152, where we do a fast-path check whether a cache/marker file exists on disk, and put the complex code inside the conditional so it only runs in the slow path?

Reply:
I'm going to try to make that clearer from the code, but it already goes first through a fast path, then through a slower one. The fast path goes from the beginning of the `static {` block up to `File jansiLib = …`. It calls a method on `coursier.paths.CachePath`, but it's a fast thing, that shouldn't load much more coursier classes (the method is [here](https://github.com/coursier/coursier/blob/8e864315c6879cbccab964323eedf21251906a95/modules/paths/src/main/java/coursier/paths/CachePath.java#L120), and it loads [that class](https://github.com/coursier/coursier/blob/8e864315c6879cbccab964323eedf21251906a95/modules/paths/src/main/java/coursier/paths/CoursierPaths.java), which itself loads [directories-jvm](https://github.com/dirs-dev/directories-jvm) classes - it's all pure Java classes).

Reply:
Yes I'm aware it currently goes through all lightweight JVM classes, and the manual startup-time benchmarks verify it. But we should refactor the code to make it obvious at a glance rather than having to dig through third-party libraries, otherwise it is prone to regress when someone overlooks it and makes changes later

Reply:
Just put the loading logic aside in a separate class, with two methods, `tryLoadFast` and `loadSlow` (did you mean that kind of thing?)